crates/sui-network/src/state_sync/mod.rs

```diff
@@ -418,28 +424,30 @@ where
                 assert_eq!(checkpoint.previous_digest(), Some(previous_digest));
             }
 
+            debug!("consensus sent too new of a checkpoint");
+
             // See if the missing checkpoints are already in our store and quickly update our
             // watermarks
-            for sequence_number in next_sequence_number..=checkpoint.sequence_number() {
-                if let Some(checkpoint) = self
-                    .store
-                    .get_checkpoint_by_sequence_number(sequence_number)
-                    .expect("store operation should not fail")
-                {
-                    self.store
-                        .insert_checkpoint(checkpoint.clone())
-                        .expect("store operation should not fail");
+            let mut checkpoints_from_storage =
+                (next_sequence_number..=checkpoint.sequence_number()).map(|n| {
```

Comment:
we could just break in a `else` of `if let`, not big deal anyways!

Reply:
yeah true, this has the same effect too :)